src/v/storage/segment_utils.cc

```diff
@@ -499,24 +499,28 @@ ss::future<std::optional<size_t>> do_self_compact_segment(
 
 ss::future<> rebuild_compaction_index(
   model::record_batch_reader rdr,
-  ss::lw_shared_ptr<storage::stm_manager>,
-  std::vector<model::tx_range>&&,
+  ss::lw_shared_ptr<storage::stm_manager> stm_manager,
+  std::vector<model::tx_range>&& aborted_txs,
   std::filesystem::path p,
   compaction_config cfg,
   storage_resources& resources) {
     return make_compacted_index_writer(p, cfg.sanitize, cfg.iopc, resources)
-      .then([r = std::move(rdr)](compacted_index_writer w) mutable {
+      .then([r = std::move(rdr), stm_manager, txs = std::move(aborted_txs), p](
+              compacted_index_writer w) mutable {
           auto u = std::make_unique<compacted_index_writer>(std::move(w));
           auto ptr = u.get();
           return std::move(r)
-            .consume(index_rebuilder_reducer(ptr), model::no_timeout)
-            .then_wrapped([x = std::move(u)](ss::future<> fut) mutable {
+            .consume(
+              tx_reducer(stm_manager, std::move(txs), ptr), model::no_timeout)
+            .then_wrapped([x = std::move(u),
+                           p](ss::future<tx_reducer::stats> fut) mutable {
+                vlog(gclog.info, "tx reducer path: {} stats {}", p, fut.get0());
                 return x->close()
                   .handle_exception([](std::exception_ptr e) {
                       vlog(gclog.warn, "error closing compacted index:{}", e);
                   })
                   .then([f = std::move(fut), x = std::move(x)]() mutable {
-                      return std::move(f);
+                      return ss::make_ready_future<>();
```

Comment:
This `then` can be replaced with `finally` just to capture the `x` variable

Reply:
nit the ` return ss::make_ready_future<>();` is redundant it can be an empty lambda, also it doesn't have to be mutable